### ISSUE TITLE
I meet one problem: when my cert expire and the connection does not emit an error?

### ISSUE THREAD

**Issue by zhhxu2011** [argon/node-apn#292]
I want to know why the connection does not emit an error when my cert expire?
And only disconnect event was emitted, and socket.length is 0.
Wish anyone can help!

**Comment by argon** [comment]
Are you using the latest version of node-apn?

―A

**Comment by zhhxu2011** [comment]
yes, the latest version

**Comment by argon** [comment]
node-apn will check the certificate validity when it is first loaded but only once. If the process is running for a long time then it will not detect failures after the certificate expires. This is something I may consider adding.

**Comment by argon** [comment]
This is no longer relevant as everyone should use the new [Provider Authentication Token](https://developer.apple.com/library/prerelease/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/APNsProviderAPI.html#//apple_ref/doc/uid/TP40008194-CH101-SW21) support in v2.1.0
